### PR TITLE
Assorted Changes 

### PR DESCRIPTION
**Pull request by Legorooj**

### CODE REVIEW

File: modulegraph2/_modulegraph.py [hunk 1]
```diff
@@ -79,11 +79,9 @@ def __init__(
         self._missing_hook = FirstNotNone()
         self._work_stack = []
 
-        self._global_lazy_nodes = {}
-
         # Reference to __main__ cannot be valid when multiple scripts
         # are added to the graph, just ignore this module for now.
-        self._global_lazy_nodes["__main__"] = None
```
Comment (ronaldoussoren) [inline modulegraph2/_modulegraph.py:86]: I don't like this change.

File: modulegraph2/_modulegraph.py [hunk 2]
```diff
@@ -669,10 +667,9 @@ def _find_or_load_module(
         Returns:
           The node for *importing_module*.
         """
-        node: Optional[BaseNode] = None
         parent_node: Optional[BaseNode]
 
-        node = self._find_module(module_name)
+        node: Optional[BaseNode] = self._find_module(module_name)
```
Comment (ronaldoussoren) [inline modulegraph2/_modulegraph.py:672]: why is this necessary?  ``Optional[BaseNode]`` is the return type of ``_find_module``.

**Review by ronaldoussoren — CHANGES_REQUESTED** [review]

File: modulegraph2/_modulegraph.py [hunk 3]
```diff
@@ -450,8 +448,8 @@ def _load_module(
         """
         node: BaseNode
 
-        assert not module_name.startswith(".")
-        assert self.find_node(module_name) is None
+        assert not module_name.startswith("."), f'module_name must not start with a "."'
+        assert self.find_node(module_name) is None, f'"{module_name}" is already in the graph'
```
Comment (ronaldoussoren) [inline modulegraph2/_modulegraph.py:453]: I don't mind this change, but have you found a way to trigger these asserts through the public API?

File: modulegraph2/_modulegraph.py (hunk 1 above)
Comment (Legorooj) [reply modulegraph2/_modulegraph.py:86]: May I ask why not?

File: modulegraph2/_modulegraph.py (hunk 3 above)
Comment (Legorooj) [reply modulegraph2/_modulegraph.py:453]: Not these 2 asserts (I just saw them and added the messages), though I've found issues on some of the other assert statements. Taking a look at them myself first before I submit an issue.
